### PR TITLE
Remove test reset check for GCU test

### PR DESCRIPTION
**Pull request by wen587**
### Description of PR

Summary: Remove test reset check for GCU test cause GCU feature will take care of it
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Dhcp_relay test: Remove reset check and rollback instead of reload. Use MACRO for each setup. Add try catch for each test to ensure checkpoint is removed after test.
Syslog test: Rename function. Setup code restructure. No logical change.
#### How did you do it?
Remove code related
#### How did you verify/test it?
Run test of sonic-mgmt/tests/generic_config_updater/test_dhcp_relay.py on KVM

### CODE REVIEW

File: tests/generic_config_updater/test_dhcp_relay.py [hunk 1]
```diff
@@ -188,7 +219,7 @@ def dhcp_severs_by_vlanid(duthost, vlanid):
     root          73       1  0 06:39 pts/0    00:00:00 /usr/sbin/dhcrelay -d -m discard -a %h:%p %P --name-alias-map-file /tmp/port-name-alias-map.txt -id Vlan1000 -iu Vlan100 -iu PortChannel0001 -iu PortChannel0002 -iu PortChannel0003 -iu PortChannel0004 192.0.0.1 192.0.0.2 192.0.0.3 192.0.0.4
     """
     cmds = "docker exec dhcp_relay supervisorctl status | grep ^dhcp-relay \
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_dhcp_relay.py:221]: >grep [](http://example.com/codeflow?start=58&length=4)

combine 2 grep into one?

File: tests/generic_config_updater/test_syslog.py [hunk 2]
```diff
@@ -116,10 +116,13 @@ def setup_env(duthosts, rand_one_dut_hostname, cfg_facts, init_syslog_config, or
     duthost = duthosts[rand_one_dut_hostname]
     create_checkpoint(duthost)
 
-    syslog_config_cleanup(duthost, cfg_facts)
-
-    if init_syslog_config == CONFIG_ADD_DEFAULT:
+    if init_syslog_config == CONFIG_CLEANUP:
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_syslog.py:119]: >CONFIG_CLEANUP [](http://example.com/codeflow?start=29&length=14)

There are lots of code changes not listed in PR description. Could you list all the motivations?

We would like to understand if any is refactoring, or fixing something.

File: tests/generic_config_updater/test_dhcp_relay.py (hunk 1 above)
Comment (wen587) [reply tests/generic_config_updater/test_dhcp_relay.py:221]: Will do

File: tests/generic_config_updater/test_syslog.py (hunk 2 above)
Comment (wen587) [reply tests/generic_config_updater/test_syslog.py:119]: Updated PR motivation. Please have a look. Thanks

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**